File: Java/dependency-injection/service-locator/src/main/java/ServiceLocatorAntipattern/InMessages/StopMessage.java
package ServiceLocatorAntipattern.InMessages;

import ServiceLocatorAntipattern.Interfaces.AcmeMessage;
import ServiceLocatorAntipattern.Interfaces.DataDestination;
import ServiceLocatorAntipattern.Services.IAuthorization;

public class StopMessage implements AcmeMessage {
  public void authorizeUsing(IAuthorization authorizationRules) {
    System.out.println("Authorizing Stop with " + authorizationRules);
  }

  public void writeTo(DataDestination dataDestination) {
    System.out.println("Writing Stop to " + dataDestination);
  }
}

File: CSharp/DomainSpecificLanguages/DomainSpecificLanguages/AlarmsNestedFunctions/InMessages/StopMessage.cs
using System;
using TelecomSystemNestedFunctions.Interfaces;
using TelecomSystemNestedFunctions.Services;

namespace TelecomSystemNestedFunctions.InMessages
{
  class StopMessage : AcmeMessage
  {
    public void AuthorizeUsing(IAuthorization authorizationRules)
    {
      Console.WriteLine("Authorizing Stop with " + authorizationRules);
    }

    public void WriteTo(DataDestination dataDestination)
    {
      Console.WriteLine("Writing Stop to " + dataDestination);
    }
  }
}